### DISCUSSION TITLE
Beginners Microscope with Electronics

### DISCUSSION THREAD

**Discussion by wbrett87** [bionanoimaging/UC2-GIT#94] [Q&A]
Hey so I have most of the scope in your beginners tutorial built. I have a couple modifications:

1) I'm using an arducam hawkeye 64mp camera instead of cell phone camera
2) I have the motor and light connected to seperate esp32s and i'd like to integrate them wirelessly with imswitch. Is this possible with this scope?

So really the reason for this post is to ask if I can integrate all the control(camera,motorized focus, and light) of this basic microscope with imswitch?

**Comment by beniroquai** [comment]
In principle this should be possible. Is the hawkey camera connected via USB and can be addressed via OpenCVs driver? 
Any reason why you want the ESP to be connected wirelessly? 
Do you already have a picture of your setup? I'm very curious! :-)

**Reply by wbrett87** [reply]
To be honest I don't really need wireless connection on this scope, but thought it would be nice to learn so I could apply it to the incubator scope I plan on building in the future. Do you think it would be easier via serial connection? The hawkeye camera is CSI2 but I have an adapter to connect via usb. Not sure about OpenCVs, I know it can use the rlibcamera software available on raspberry pis. I plan on running IMswitch on a raspberry pi 4b. I'll send pics soon!

**Comment by beniroquai** [comment]
We can learn together. Haven't touched the wifi part for the uc2-REST
framework for a while now. USB is definitely easier to setup (eg detection
of serial devices)
